awx/templates/deployment.yaml

```diff
@@ -1,28 +1,41 @@
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ include "awx.fullname" . }}-web
+  name: {{ include "awx.fullname" . }}
   labels:
-    app.kubernetes.io/name: {{ include "awx.name" . }}-web
+    app.kubernetes.io/name: {{ include "awx.name" . }}
     helm.sh/chart: {{ include "awx.chart" . }}
     app.kubernetes.io/instance: {{ .Release.Name }}
     app.kubernetes.io/managed-by: {{ .Release.Service }}
 spec:
-  replicas: {{ .Values.web.replicaCount }}
+  replicas: {{ .Values.replicaCount }}
   selector:
     matchLabels:
-      app.kubernetes.io/name: {{ include "awx.name" . }}-web
+      app.kubernetes.io/name: {{ include "awx.name" . }}
       app.kubernetes.io/instance: {{ .Release.Name }}
   template:
     metadata:
       labels:
-        app.kubernetes.io/name: {{ include "awx.name" . }}-web
+        app.kubernetes.io/name: {{ include "awx.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
     spec:
+      serviceAccountName: {{ include "awx.serviceAccountName" $ }}
       containers:
-        - name: awx
-          image: "{{ .Values.web.image.repository }}:{{ .Values.web.image.tag }}"
-          imagePullPolicy: {{ .Values.web.image.pullPolicy }}
+        - name: task
```

Comment:
Does this mean we cannot task/worker processes independently of web processes?

Reply:
That's correct. Originally, the ansible AWX team had web, worker, memcached, and rabbitmq all running in a single pod (statefulset). The only distributed service was postgres. They eventually switched to a Deployment, and the rabbitmq, memcached services were dropped in favor of redis, but the same is still true. Web, worker, and redis all run in the same pod. 

What I noticed is that when running them separately, web has repeated failures trying to establish a websocket connection between it and the worker. If you look at the nginx config that's used in the container, it listens on a separate port for websockets—a port that isn't exposed via a service at all. Apparently there's also hostname resolution things going on as well that don't seem to work great out of the box in kubernetes. I think all of these things are the reasons web/worker run as a single unit. Since the official installer does it this way and I was getting persistent errors in the logs, I set it up the same way and everything was working again.